pkg/apis/edge/v1alpha1/associate.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2023 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// AssociatorAnnotationKey is the key of an annotation used to point from a workload
+// object to an Associator that says how associated objects in edge clusters are identified.
+// For a namespaced object, the value of the annotation is the name of the Associator.
+// For a non-namespaced object, the value of the annotation is the namespace/name of the Associator.
+// When the annotation is absent, no associated objects are identified.
+const AssociatorAnnotationKey = "edge.kcp.io/associator"
+
```

Comment:
Have you considered how the current kcp TMC syncer up-syncing works, so that we can take advantages of features that it already provides? See for example https://github.com/kcp-dev/kcp/issues/1980

Reply:
No, I do not understand the thoughts there.  I asked yesterday (https://kubernetes.slack.com/archives/C021U8WSAFK/p1678826090217939) and got no answer.  Thanks for the pointer.  I am also pursuing updates to https://github.com/kcp-dev/kcp/blob/e1353f4e70eecfcab0ddb124563f1ac8e0e0aca1/docs/content/concepts/placement-locations-and-scheduling.md